themes/bootstrap3/templates/Recommend/CatalogResults.phtml

```diff
@@ -1,9 +1,10 @@
 <?php
   $searchObject = $this->recommend->getResults();
   $results = $searchObject->getResults();
+  $heading = $this->recommend->getCustomHeading() ?? 'Catalog Results';
 ?>
 <?php if (!empty($results)): ?>
-  <h4><?=$this->transEsc('Catalog Results')?></h4>
+  <h4><?=$this->transEsc($heading)?></h4>
```

Comment:
The custom heading bit could be extracted to a separate PR if needed.

Reply:
I don't mind adding these two enhancements in a single PR, though maybe we should rename the PR to something broader if we do.

Reply:
Done